msbuild/Xamarin.Shared/Xamarin.Shared.targets

```diff
@@ -595,10 +596,62 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 		 to the CollectAppManifestsDependsOn property. -->
 	<Target Name="CollectAppManifests" DependsOnTargets="$(CollectAppManifestsDependsOn)" />
 
-	<!-- This target has no inputs, because it must still be run if there are no input app manifests, to set default values -->
+	<Target Name="_CompileAppManifestInputs"
+		Condition="'$(IsMacEnabled)' == 'true' Or '$(IsHotRestartBuild)' == 'true'"
+		>
+		<PropertyGroup>
+			<_CompileAppManifestInputsFile>$(DeviceSpecificIntermediateOutputPath)_CompileAppManifest.inputs</_CompileAppManifestInputsFile>
+			<_CompileAppManifestFontFilesToRegister>@(_FontFilesToRegister)</_CompileAppManifestFontFilesToRegister>
+			<_CompileAppManifestPartialAppManifest>@(PartialAppManifest)</_CompileAppManifestPartialAppManifest>
+		</PropertyGroup>
+		<ItemGroup>
+			<_CompileAppManifestInputLine Include="SessionId=$(BuildSessionId)" />
+			<_CompileAppManifestInputLine Include="ApplicationId=$(ApplicationId)" />
+			<_CompileAppManifestInputLine Include="ApplicationTitle=$(ApplicationTitle)" />
+			<_CompileAppManifestInputLine Include="ApplicationVersion=$(ApplicationVersion)" />
+			<_CompileAppManifestInputLine Include="ApplicationDisplayVersion=$(ApplicationDisplayVersion)" />
+			<_CompileAppManifestInputLine Include="AppBundleName=$(_AppBundleName)" />
+			<_CompileAppManifestInputLine Include="AppManifest=$(AppBundleManifest)" />
```

Comment:
We also have to rerun the task for inputs that are files, if those files changed.

I think this could be fixed by making `_CompileAppManifestInputsFile` an item group, and adding all inputs that are files to this item group (in addition to the new file you're creating).